auth0/src/main/java/com/auth0/android/authentication/storage/CredentialsManager.java

```diff
@@ -71,15 +71,14 @@ public void getCredentials(@NonNull final BaseCallback<Credentials, CredentialsM
         String tokenType = storage.retrieve(KEY_TOKEN_TYPE);
         String expiresInValue = storage.retrieve(KEY_EXPIRES_IN);
         String expirationTimeValue = storage.retrieve(KEY_EXPIRATION_TIME);
-        Long expiresIn = Long.parseLong(expiresInValue);
 
-        boolean invalidTokens = isEmpty(accessToken) && isEmpty(idToken);
-        if (invalidTokens) {
-            callback.onFailure(new CredentialsManagerException("No Credentials where previously set."));
+        if (isEmpty(accessToken) && isEmpty(idToken) || isEmpty(expiresInValue) || isEmpty(expirationTimeValue)) {
+            callback.onFailure(new CredentialsManagerException("No Credentials were previously set."));
             return;
         }
-        long expirationTime = expirationTimeValue == null ? 0 : Long.parseLong(expirationTimeValue);
+        long expirationTime = Long.parseLong(expirationTimeValue);
         if (expirationTime > getCurrentTimeInMillis()) {
+            Long expiresIn = Long.parseLong(expiresInValue);
```

Comment:
`expiresIn` value is not right anymore in this case. you can use the expirationTimeValue - currentTime if necessary.